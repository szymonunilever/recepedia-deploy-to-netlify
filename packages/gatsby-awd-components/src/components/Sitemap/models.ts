import { TagName } from '../Text';
import { SitemapCategoryEntry } from './partials';

export interface SitemapProps {
  title?: string;
  titleLevel?: TagName;
  content: SitemapCategoryEntry[];
  wrapBlocks?: boolean;
}
