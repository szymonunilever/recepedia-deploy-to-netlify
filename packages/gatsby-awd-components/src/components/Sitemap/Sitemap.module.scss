@import '../../scss/config';
.sitemap {
  :global {
    .sitemap {
      &__categories {
        display: flex;
        flex-direction: column;
        @include respondTo(medium) {
          flex-direction: row;
        }

        & > .sitemap__category {
          @include respondTo(medium) {
            text-align: left;
            box-sizing: border-box;
            flex: 1;
          }
        }
      }

      &__category {
        &-title {
          font-size: $sitemap-category-heading-fz;
        }
        &-items {
          a {
            font-size: $sitemap-subheading-fz;
          }
        }
      }
      &__block {
        flex-basis: 100% / $sitemap-columns-number;
        @include respondTo(medium) {
          flex-basis: 100% / $sitemap-columns-number-lg;
        }
      }
      /////----------- CUSTOM -----------/////
      &__categories {
        & > .sitemap__category {
          margin: 0 0 32px 0;
          text-align: center;
          @include respondTo(medium) {
            text-align: left;
            box-sizing: border-box;
            width: calc(100% / 2);
            padding: 0 24px 0 0;
          }
          @include respondTo(desktop) {
            width: calc(99.9% / 3);
          }
        }
      }

      &__category {
        &-title {
          font-size: 1.8rem;
          margin: 0 0 10px;
          text-transform: uppercase;
          font-family: $rubik;
          font-weight: 500;
          @include respondTo(medium) {
            font-size: 2.4rem;
          }
        }

        &-items {
          a {
            line-height: 1;
            text-transform: none;
            font-weight: 500;
            @include respondTo(medium) {
              font-size: 1.4rem;
              line-height: 1.4;
            }
          }
        }
      }
    }
    a {
      line-height: 1.4;
      font-family: $rubik;
      font-weight: 500;
      padding-right: 20px;
    }
    /////----------- CUSTOM END -----------/////
    a {
      display: block;
      text-decoration: none;
      cursor: pointer;
      color: $sitemap-link-color;
      transition: $sitemap-link-transition;
      &:hover {
        color: $sitemap-link-hover-color;
      }
    }

    ul {
      list-style: none;
      padding: 0;
    }
  }
}
