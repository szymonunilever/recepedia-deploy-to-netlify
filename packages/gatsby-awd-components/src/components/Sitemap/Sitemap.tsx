import React from 'react';
import { SitemapProps } from './models';
import { SitemapCategoryEntry } from "./partials";
import { Text } from '../Text';
import { TagName } from "../Text";
import { SitemapCategory } from './partials';
import partition from 'lodash/partition';
import cx from 'classnames';
import theme from './Sitemap.module.scss';

const Sitemap = ({
  title,
  titleLevel = TagName.h1,
  content,
  wrapBlocks = false,
}: SitemapProps) => {
  const categoryListsArray = wrapBlocks
    ? partition(
        content,
        item => item.categoryItems && item.categoryItems.length
      )
    : [content];

  return (
    <div className={cx(theme.sitemap, 'sitemap')}>
      {title && (
        <Text tag={titleLevel} text={title} className={'sitemap-title'} />
      )}
      <ul className="sitemap__categories">
        {categoryListsArray[0].map(
          (item: SitemapCategoryEntry, index: number) =>
            wrapBlocks ? (
              <span className="sitemap__block" key={index}>
                <SitemapCategory key={index} item={item} />
              </span>
            ) : (
              <SitemapCategory key={index} item={item} />
            )
        )}
        {wrapBlocks && (
          <div className="sitemap__block">
            {categoryListsArray[1].map(
              (item: SitemapCategoryEntry, index: number) => (
                <SitemapCategory key={index} item={item} />
              )
            )}
          </div>
        )}
      </ul>
    </div>
  );
};

export default Sitemap;
