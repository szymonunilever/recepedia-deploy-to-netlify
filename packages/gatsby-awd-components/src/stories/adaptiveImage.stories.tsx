import { storiesOf } from '@storybook/react';
import React from 'react';
import { AdaptiveImage } from '../index';
// import localImage from './assets/localImage';
import localImage from './assets/localImageMulti';
// import { findLastIndex } from 'lodash-es';

const image = {
  url:
    'https://scm-assets.constant.co/scm/unilever/1d398653b55393fd6da9bff8ea193338/bab4fc1b-c269-44c1-8d60-367626f8b029.jpg',
  alt: 'Alt',
};

storiesOf('Generic/AdaptiveImage', module)
  .add(
    'default',
    () => (
      <AdaptiveImage
        className="custom-class"
        localImage={localImage}
        alt={image.alt}
      />
    ),
    { inline: false }
  )
  .add('with link', () => (
    <AdaptiveImage
      className="custom-class"
      localImage={localImage}
      alt={image.alt}
      link={'/'}
    />
  ))
  .add('with sizes specified', () => (
    <div>
      <AdaptiveImage
        className="custom-class"
        localImage={localImage}
        alt={image.alt}
        sizes="(max-width: 320px) 100vw, 200px"
      />
      <p>
        Note: image width is set to 200px. If browser cache has the same image
        of bigger size, it will be used instead of small image.
      </p>
    </div>
  ));
