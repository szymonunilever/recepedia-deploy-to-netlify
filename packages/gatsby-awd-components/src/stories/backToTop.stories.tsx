import { storiesOf } from '@storybook/react';
import React from 'react';
import { ReactComponent as Icon } from 'src/svgs/inline/arrow-up.svg';
import { BackToTop } from '../index';
import { ButtonViewType } from '../components/Button';

const page = {
  height: '200vh',
  fontSize: '2rem',
};

const content: AppContent.BackToTopContent = {
  label: 'Back to top',
};

storiesOf('Generic/Back To Top', module)
  .add(
    'with Icon',
    () => (
      <>
        <div style={page}>
          Scroll down to see Back-to-top button in the right bottom corner
          <BackToTop content={{ label: undefined }} Icon={Icon} />
        </div>
      </>
    ),
    { inline: false }
  )
  .add(
    'with Icon and Text',
    () => (
      <>
        <div style={page}>
          Scroll down to see Back-to-top button in the right bottom corner
          <BackToTop content={content} Icon={Icon} />
        </div>
      </>
    ),
    { inline: false }
  )
  .add(
    'with Icon viewType',
    () => (
      <>
        <div style={page}>
          Scroll down to see Back-to-top button in the right bottom corner
          <BackToTop
            content={{ label: undefined }}
            Icon={Icon}
            viewType={ButtonViewType.icon}
          />
        </div>
      </>
    ),
    { inline: false }
  );
