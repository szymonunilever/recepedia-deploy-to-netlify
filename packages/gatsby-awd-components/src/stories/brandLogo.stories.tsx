import { storiesOf } from '@storybook/react';
import React from 'react';
import { BrandLogo } from '../components/BrandLogo';

const brandList = ['knorr', 'hellmanns', 'maizena'];

storiesOf('Generic/BrandLogo', module).add('Default view', () => (
  <>
    {brandList.map(brand => (
      <BrandLogo brandTheme={brand} linkTo={'/'} key={brand} />
    ))}
  </>
));
