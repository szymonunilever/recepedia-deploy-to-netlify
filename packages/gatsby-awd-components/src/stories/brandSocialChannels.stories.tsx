import { storiesOf } from '@storybook/react';
import React from 'react';
import { BrandSocialChannels } from '../index';
import content from '../mocks/brandSocialChannels.json';
import { ReactComponent as FbIcon } from 'src/svgs/inline/facebook.svg';
import { ReactComponent as TwitterIcon } from 'src/svgs/inline/twitter.svg';
import { ReactComponent as InstaIcon } from 'src/svgs/inline/instagram.svg';

const list = {
  facebook: <FbIcon />,
  twitter: <TwitterIcon />,
  instagram: <InstaIcon />,
};

const config = {
  displayLabel: false,
};

storiesOf('Generic/BrandSocialChannels', module).add('default', () => (
  <BrandSocialChannels listIcons={list} content={content} {...config} />
));
