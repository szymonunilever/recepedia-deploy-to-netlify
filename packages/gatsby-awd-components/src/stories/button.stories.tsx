import { storiesOf } from '@storybook/react';
import React from 'react';
import { Button, ButtonViewType } from '../index';
import { ReactComponent as Facebook } from '../svgs/inline/facebook.svg';
import { ReactComponent as Facebook2 } from '../svgs/inline/facebook2.svg';

const onClick = () => alert('on click button action');

storiesOf('Generic/Button', module)
  .add('with text label', () => (
    <>
      <Button {...{ onClick }} className="button--small">
        small
      </Button>
      <Button {...{ onClick }} className="button--medium">
        medium
      </Button>
      <Button {...{ onClick }} className="button--large">
        large
      </Button>
    </>
  ))
  .add('with text label disabled', () => (
    <Button {...{ onClick }} className="custom-class" isDisabled>
      Click me
    </Button>
  ))
  .add('with icon viewType', () => (
    <div style={{ fontSize: '3rem' }}>
      <Button
        {...{ onClick }}
        Icon={Facebook}
        className="custom-class"
        viewType={ButtonViewType.icon}
      />
    </div>
  ))
  .add('with different icons for toggled states', () => (
    <div style={{ fontSize: '3rem' }}>
      <Button
        Icon={Facebook}
        IconSelected={Facebook2}
        className="custom-class"
        isToggle
        viewType={ButtonViewType.icon}
      />
    </div>
  ))
  .add('with icon and text', () => (
    <Button Icon={Facebook} {...{ onClick }} className="custom-class">
      Click me
    </Button>
  ));
