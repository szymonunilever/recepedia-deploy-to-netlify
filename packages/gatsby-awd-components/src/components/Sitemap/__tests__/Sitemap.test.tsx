import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import Sitemap, { SitemapProps } from '../index';
import toJson from 'enzyme-to-json';
import { content } from '../../../stories/sitemap.stories';

describe('<Sitemap />', () => {
  let wrapper: ReactWrapper;
  const sitemapProps: SitemapProps = {
    content,
  };

  beforeEach(() => {
    wrapper = mount(<Sitemap {...sitemapProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
