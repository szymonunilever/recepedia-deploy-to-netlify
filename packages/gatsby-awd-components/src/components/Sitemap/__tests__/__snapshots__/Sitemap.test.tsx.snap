// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sitemap /> matches the snapshot 1`] = `
<Sitemap
  content={
    Array [
      Object {
        "categoryItems": Array [
          Object {
            "path": "/recipes/recipe1",
            "title": "recipe1",
          },
          Object {
            "path": "/recipes/recipe2",
            "title": "recipe2",
          },
          Object {
            "path": "/recipes/recipe3",
            "title": "recipe3",
          },
          Object {
            "path": "/recipes/recipe4",
            "title": "recipe4",
          },
          Object {
            "path": "/recipes/recipe5",
            "title": "recipe5",
          },
          Object {
            "path": "/recipes/recipe6",
            "title": "recipe6",
          },
          Object {
            "path": "/recipes/recipe7",
            "title": "recipe7",
          },
        ],
        "path": "/recipes",
        "title": "Recipes",
      },
      Object {
        "categoryItems": Array [
          Object {
            "path": "/recipes/product1",
            "title": "product1",
          },
          Object {
            "path": "/recipes/product2",
            "title": "product2",
          },
          Object {
            "path": "/recipes/product3",
            "title": "product3",
          },
          Object {
            "path": "/recipes/product4",
            "title": "product4",
          },
          Object {
            "path": "/recipes/product5",
            "title": "product5",
          },
          Object {
            "path": "/recipes/product6",
            "title": "product6",
          },
          Object {
            "path": "/recipes/product7",
            "title": "product7",
          },
        ],
        "path": "/products",
        "title": "Products",
      },
      Object {
        "path": "/contacts",
        "title": "Contact us",
      },
      Object {
        "path": "/faq",
        "title": "FAQ",
      },
    ]
  }
>
  <div
    className="sitemap sitemap"
  >
    <ul
      className="sitemap__categories"
    >
      <SitemapCategory
        item={
          Object {
            "categoryItems": Array [
              Object {
                "path": "/recipes/recipe1",
                "title": "recipe1",
              },
              Object {
                "path": "/recipes/recipe2",
                "title": "recipe2",
              },
              Object {
                "path": "/recipes/recipe3",
                "title": "recipe3",
              },
              Object {
                "path": "/recipes/recipe4",
                "title": "recipe4",
              },
              Object {
                "path": "/recipes/recipe5",
                "title": "recipe5",
              },
              Object {
                "path": "/recipes/recipe6",
                "title": "recipe6",
              },
              Object {
                "path": "/recipes/recipe7",
                "title": "recipe7",
              },
            ],
            "path": "/recipes",
            "title": "Recipes",
          }
        }
        key="0"
      >
        <mockConstructor
          className="sitemap__category-title"
          to="/recipes"
        >
          <a
            className="sitemap__category-title"
            href="/recipes"
          >
            Recipes
          </a>
        </mockConstructor>
        <ul
          className="sitemap__category-items"
        >
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/recipe1",
                "title": "recipe1",
              }
            }
            key="0"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/recipe1"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/recipe1"
              >
                recipe1
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/recipe2",
                "title": "recipe2",
              }
            }
            key="1"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/recipe2"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/recipe2"
              >
                recipe2
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/recipe3",
                "title": "recipe3",
              }
            }
            key="2"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/recipe3"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/recipe3"
              >
                recipe3
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/recipe4",
                "title": "recipe4",
              }
            }
            key="3"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/recipe4"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/recipe4"
              >
                recipe4
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/recipe5",
                "title": "recipe5",
              }
            }
            key="4"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/recipe5"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/recipe5"
              >
                recipe5
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/recipe6",
                "title": "recipe6",
              }
            }
            key="5"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/recipe6"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/recipe6"
              >
                recipe6
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/recipe7",
                "title": "recipe7",
              }
            }
            key="6"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/recipe7"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/recipe7"
              >
                recipe7
              </a>
            </mockConstructor>
          </SitemapCategory>
        </ul>
      </SitemapCategory>
      <SitemapCategory
        item={
          Object {
            "categoryItems": Array [
              Object {
                "path": "/recipes/product1",
                "title": "product1",
              },
              Object {
                "path": "/recipes/product2",
                "title": "product2",
              },
              Object {
                "path": "/recipes/product3",
                "title": "product3",
              },
              Object {
                "path": "/recipes/product4",
                "title": "product4",
              },
              Object {
                "path": "/recipes/product5",
                "title": "product5",
              },
              Object {
                "path": "/recipes/product6",
                "title": "product6",
              },
              Object {
                "path": "/recipes/product7",
                "title": "product7",
              },
            ],
            "path": "/products",
            "title": "Products",
          }
        }
        key="1"
      >
        <mockConstructor
          className="sitemap__category-title"
          to="/products"
        >
          <a
            className="sitemap__category-title"
            href="/products"
          >
            Products
          </a>
        </mockConstructor>
        <ul
          className="sitemap__category-items"
        >
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/product1",
                "title": "product1",
              }
            }
            key="0"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/product1"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/product1"
              >
                product1
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/product2",
                "title": "product2",
              }
            }
            key="1"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/product2"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/product2"
              >
                product2
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/product3",
                "title": "product3",
              }
            }
            key="2"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/product3"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/product3"
              >
                product3
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/product4",
                "title": "product4",
              }
            }
            key="3"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/product4"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/product4"
              >
                product4
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/product5",
                "title": "product5",
              }
            }
            key="4"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/product5"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/product5"
              >
                product5
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/product6",
                "title": "product6",
              }
            }
            key="5"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/product6"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/product6"
              >
                product6
              </a>
            </mockConstructor>
          </SitemapCategory>
          <SitemapCategory
            item={
              Object {
                "path": "/recipes/product7",
                "title": "product7",
              }
            }
            key="6"
          >
            <mockConstructor
              className="sitemap__category-title"
              to="/recipes/product7"
            >
              <a
                className="sitemap__category-title"
                href="/recipes/product7"
              >
                product7
              </a>
            </mockConstructor>
          </SitemapCategory>
        </ul>
      </SitemapCategory>
      <SitemapCategory
        item={
          Object {
            "path": "/contacts",
            "title": "Contact us",
          }
        }
        key="2"
      >
        <mockConstructor
          className="sitemap__category-title"
          to="/contacts"
        >
          <a
            className="sitemap__category-title"
            href="/contacts"
          >
            Contact us
          </a>
        </mockConstructor>
      </SitemapCategory>
      <SitemapCategory
        item={
          Object {
            "path": "/faq",
            "title": "FAQ",
          }
        }
        key="3"
      >
        <mockConstructor
          className="sitemap__category-title"
          to="/faq"
        >
          <a
            className="sitemap__category-title"
            href="/faq"
          >
            FAQ
          </a>
        </mockConstructor>
      </SitemapCategory>
    </ul>
  </div>
</Sitemap>
`;
